src/screens/application/hook/useInitApplication.tsx

```diff
@@ -66,13 +78,148 @@ export const useInitApplication = () => {
+    if(Platform.OS === 'android'){
+      messagingEventRef.current = messaging().onNotificationOpenedApp((remoteMessage)=>{
+        _pushNavigate(remoteMessage);
+      })
+
+      const initialNotification = await messaging().getInitialNotification()
+      if(initialNotification){
+        _pushNavigate(initialNotification)
+      }
+    }else if (Platform.OS === 'ios'){
+      //for ios, notifee events work while messaging event are malfunctioning, the foreground event 
+      // on ios is called if user opens/starts app from notification
+      notifeeEventRef.current = notifee.onForegroundEvent(({type, detail})=>{
+        if(type === EventType.PRESS){
+          _pushNavigate(detail.notification);
+        }
+      })
+  
+    }
+   
+    
+    
+  }
+
+
   const _handleAppStateChange = (nextAppState) => {
     if (appState.current.match(/inactive|background/) && nextAppState === 'active') {
       userActivityMutation.lazyMutatePendingActivities();
     }
 
     appState.current = nextAppState;
   };
+
+  const _pushNavigate = (notification) => {
+
+    let params = null;
+    let key = null;
+    let routeName = null;
+
+    if (notification) {
+      const push = get(notification, 'data');
+      const type = get(push, 'type', '');
+      const fullPermlink =
+        get(push, 'permlink1', '') + get(push, 'permlink2', '') + get(push, 'permlink3', '');
+      // const username = get(push, 'target', '');
+      const activity_id = get(push, 'id', '');
+
+      switch (type) {
+        case 'vote':
+        case 'unvote':
+          params = {
+            author: get(push, 'target', ''),
+            permlink: fullPermlink,
+          };
+          key = fullPermlink;
+          routeName = ROUTES.SCREENS.POST;
+          break;
+        case 'mention':
+          params = {
+            author: get(push, 'source', ''),
+            permlink: fullPermlink,
+          };
+          key = fullPermlink;
+          routeName = ROUTES.SCREENS.POST;
+          break;
+
+        case 'follow':
+        case 'unfollow':
+        case 'ignore':
+          params = {
+            username: get(push, 'source', ''),
+          };
+          key = get(push, 'source', '');
+          routeName = ROUTES.SCREENS.PROFILE;
+          break;
+
+        case 'reblog':
+          params = {
+            author: get(push, 'target', ''),
+            permlink: fullPermlink,
+          };
+          key = fullPermlink;
+          routeName = ROUTES.SCREENS.POST;
+          break;
+
+        case 'reply':
+          params = {
+            author: get(push, 'source', ''),
+            permlink: fullPermlink,
+          };
+          key = fullPermlink;
+          routeName = ROUTES.SCREENS.POST;
+          break;
+
+        case 'transfer':
+          routeName = ROUTES.TABBAR.PROFILE;
+          params = {
+            activePage: 2,
+          };
+          break;
+
+        case 'inactive':
+          routeName = ROUTES.SCREENS.EDITOR;
```

Comment:
favorite and bookmark notifications need route path as well to navigate relevant page